File: Examples/CSharp/NoteBook/ConvertToPDFAsFlattened.cs
using System.IO;
using Aspose.Note;
using Aspose.Note.Saving;
using System;
using System.Collections.Generic;

namespace Aspose.Note.Examples.CSharp.WorkingWithNoteBook
{
    public class ConvertToPDFAsFlattened
    {
        public static void Run()
        {
            // ExStart:ConvertToPDFAsFlattened
            // ExFor:Notebook
            // ExFor:NotebookSaveOptions
            // ExFor:NotebookSaveOptions.Flatten
            // ExFor:NotebookImageSaveOptions

            // ExSummary:Shows how to save flattened notebook in pdf format.

            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_NoteBook();

            // Load a OneNote Notebook
            var notebook = new Notebook(dataDir + "Notizbuch öffnen.onetoc2");

            // Save the Notebook
            dataDir = dataDir + "ConvertToPDFAsFlattened_out.pdf";
            notebook.Save(
                dataDir,
                new NotebookPdfSaveOptions
                {
                    Flatten = true
                });         

            // ExEnd:ConvertToPDFAsFlattened

            Console.WriteLine("\nNoteBook document converted to pdf as flattened successfully.\nFile saved at " + dataDir);
        }
    }
}

File: Examples/VisualBasic/NoteBook/ConvertToPDFAsFlattened.vb
Imports System.IO
Imports Aspose.Note
Imports Aspose.Note.Saving
Imports System.Collections.Generic

Namespace WorkingWithNoteBook
    Public Class ConvertToPDFAsFlattened
        Public Shared Sub Run()
            ' ExStart:ConvertToPDFAsFlattened
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir_NoteBook()

            ' Load a OneNote Notebook
            Dim notebook = New Notebook(dataDir & Convert.ToString("Notizbuch öffnen.onetoc2"))

            dataDir = dataDir & Convert.ToString("ConvertToPDFAsFlattened_out_.pdf")
            ' Save the Notebook
            notebook.Save(dataDir, New NotebookPdfSaveOptions() With {
                 .Flatten = True
            })

            ' ExEnd:ConvertToPDFAsFlattened
            Console.WriteLine(Convert.ToString(vbLf & "NoteBook document converted to pdf as flattened successfully." & vbLf & "File saved at ") & dataDir)
        End Sub
    End Class
End Namespace